### DISCUSSION TITLE
Verbindungsaufbau mit WLAN (Speedport Smart 3) nur sehr selten erfolgreich

### DISCUSSION THREAD

**Discussion by schnickes1** [tbnobody/OpenDTU#417] [Q&A] [answered]
Hallo an alle,

zunächst möchte ich mich sehr für die Bereitstellung von OpenDTU bedanken. Die Installation und Konfiguration war auch für mich als Neuling in der Thematik schnell gemacht. Vielen Dank!

**Ausgangsituation**
- AZ Delivery ESP32 Dev KitC mit AZ Delivery NRF24L01+ (integrierte Antenne)
- OpenDTU Version "40203f7".
- Verbindungsdaten für WLAN korrekt hinterlegt
- WLAN Router ist ein Speedport Smart 3
- 2,4 GHz ist aktiviert
- Abstand von ESP32 zum Router minimal (< 2m).

**Problemstellung**
Die "Network Info" zeigt mir, dass das WLAN gefunden wird (Status: activated). Es wird aber keine IP vom Router vergeben (nur 0.0.0.0) und der Status springt kurz danach auf "disabled".

Sehr selten klappt es, dass eine IP angezeigt und die Verbindung aufgebaut wird (in Router GUI zu sehen).
Die Verbindung ist dann instabil.

Das Verhalten habe ich mit 3 baugleichen ESP32 festgestellt.

**Vermutung**
Ich vermute, dass es mit dem WLAN vom ESP32 zu tun hat, habe da aber keinerlei Erfahrungswerte.
Eine defekte Hardware schließe ich erstmal aus, weil 3 Geräte das gleiche Verhalten zeigen.

Kann es sein, dass die Verbindung hardwarebedingt grundsätzlich so schlecht ist?

**Comment by Solarteur** [comment]
Das WLAN der ESP32-Module ist im Rahmen der Möglichkeiten einer "PCB-Antenne" gut. "Schräg im Haus" durch eine Betondecke ist kein Porblem. Auch werden Probleme mit der WLAN-Verbindung hier eher selten gemeldet.

Wie ist denn dein OpenDTU aufgebaut? Abstand ESP32-Moduk <---> NRF24L01+-Modul? Hast du denen Router einmal neu gestartet?

**Reply by schnickes1** [reply]
Die Verbindung ist mit Jumper Kabeln gesteckt.
Router habe ich mehrmals neugestartet.

Glaubst du, dass es an Interferenzen liegt?

**Comment by Solarteur** [comment]
Du kannst das (geflashte) ESP32-Modul ja auch ohne nRF24L01+ testen, ob so eine stabile Verbindung zu deinem WLAN möglich ist (Damit sind eventuelle Interferenzen ausgeschlossen). Was verwendest du zur Spannungsversorgung? Auch da mal probeweise einen Tausch vornehmen. 

Ergänzung: Ist der Router so eingestellt, dass er (auch) WPA2 "verarbeiten" kann?

**Reply by schnickes1** [reply]
Als Spannungsversorgung habe ich 2 verschiedene Smartphone Netzteile (5V/2A und 5V/4A) und den PC selbst schon ausprobiert.

Ich habe jetzt mal bei zweien die Stromversorgung des NRF abgezogen: Keine Verbesserung.

Router ist auf WPA2 gestellt.

Die Geräte waren ja auch schon mal korrekt verbunden. Deswegen schließe ich Konfigurationen der Geräte und Stromversorgung eigentlich aus - ansonsten wäre ja keine Verbindung zustande gekommen.

**Comment by schnickes1** [comment]
Vielleicht noch eine eher unwissenschaftliche Ergänzung:
Wenn ich in der openDTU Oberfläche die Netzwerkeinstellungen immer wieder speichere, kommt dann irgendwann mal die Verbindung zustande. 
Ich erkenne aber kein Muster.

Ich werde morgen einen Versuch mit statischer IP starten, wobei das eigentlich auch nicht Sinn der Sache ist.

**Comment by schnickes1** [comment, marked as answer]
Es hat mit fester IP geklappt.

Ich vermute jetzt, dass es dann am DHCP (des Speedports) liegt, aber da forsche ich jetzt nicht weiter.

**Comment by ipspeedport-pixel** [comment]
Das klingt nicht nach einem einzelnen defekten Gerät, sondern eher nach einem generellen Problem in der Kombination.
Wenn der ESP32 das WLAN sieht, aber keine IP bekommt (0.0.0.0), dann kommt er meist gar nicht sauber durch den DHCP-Teil vom [Speedport Smart 3](https://speedport-ips.de/). Dass es mit drei gleichen ESP32 genauso passiert, spricht ziemlich klar gegen Hardwaredefekt.
Typische Ursachen in so einem Fall:
WPA3 / gemischte Verschlüsselung am Router (ESP32 mag das oft nicht sauber)
zu strenge DHCP/Firewall-Einstellungen
2.4 GHz zwar an, aber mit “Auto-Kanal/Steering”, was manche ESPs schlecht vertragen
oder einfach instabile ESP32 WiFi-Implementierung je nach Firmware
Dass es manchmal klappt, passt auch dazu – dann erwischt er kurz eine stabile Verbindung, aber verliert sie wieder.
Kurz gesagt: eher kein “kaputt”, sondern eher ein Kompatibilitäts-/Router-Setup-Thema als echte Hardwareprobleme.
